L1Trigger/L1TTrackMatch/plugins/L1TruthTrackFastJetProducer.cc

```diff
@@ -0,0 +1,218 @@
+        if (detId.subdetId() == StripSubdetector::TOB && tTopo.tobLayer(detId) <= 3)
+          tmp_isPS = true;
+        else if (detId.subdetId() == StripSubdetector::TID && tTopo.tidRing(detId) <= 9)
+          tmp_isPS = true;
+      }
+      if (tmp_isPS)
+        trk_nPS++;
+    }
+    if (trk_nPS < trkNPSStubMin_)
+      continue;
+
+    // check that trk is real and from hard interaction
+    edm::Ptr<TrackingParticle> my_tp = MCTruthTTTrackHandle->findTrackingParticlePtr(l1track_ptr);
+    if (my_tp.isNull())  // there is no tp match so the track is fake
+      continue;
+    int tp_eventid = my_tp->eventId().event();
+    if (tp_eventid > 0)  // matched tp is from pileup
+      continue;
+
+    fastjet::PseudoJet psuedoJet(iterL1Track->momentum().x(),
+                                 iterL1Track->momentum().y(),
+                                 iterL1Track->momentum().z(),
+                                 iterL1Track->momentum().mag());
+    JetInputs.push_back(psuedoJet);                     // input tracks for clustering
+    JetInputs.back().set_user_index(this_l1track - 1);  // save track index in the collection
+  }                                                     // end loop over tracks
+
+  fastjet::ClusterSequence cs(JetInputs, jet_def);  // define the output jet collection
+  std::vector<fastjet::PseudoJet> JetOutputs =
+      fastjet::sorted_by_pt(cs.inclusive_jets(0));  // output jet collection, pT-ordered
+
+  for (unsigned int ijet = 0; ijet < JetOutputs.size(); ++ijet) {
+    math::XYZTLorentzVector jetP4(
+        JetOutputs[ijet].px(), JetOutputs[ijet].py(), JetOutputs[ijet].pz(), JetOutputs[ijet].modp());
+    float sumpt = 0;
+    float avgZ = 0;
+    std::vector<edm::Ptr<L1TTTrackType> > L1TrackPtrs;
+    std::vector<fastjet::PseudoJet> fjConstituents = fastjet::sorted_by_pt(cs.constituents(JetOutputs[ijet]));
+
+    for (unsigned int i = 0; i < fjConstituents.size(); ++i) {
+      auto index = fjConstituents[i].user_index();
+      edm::Ptr<L1TTTrackType> trkPtr(TTTrackHandle, index);
+      L1TrackPtrs.push_back(trkPtr);  // L1Tracks in the jet
+      sumpt = sumpt + trkPtr->momentum().perp();
+      avgZ = avgZ + trkPtr->momentum().perp() * trkPtr->z0();
+    }
+    avgZ = avgZ / sumpt;
+    edm::Ref<JetBxCollection> jetRef;
+    TkJet trkJet(jetP4, jetRef, L1TrackPtrs, avgZ);
+    L1TrackFastJets->push_back(trkJet);
+  }  //end loop over Jet Outputs
+
+  if (displaced_)
+    iEvent.put(std::move(L1TrackFastJets), "L1TruthTrackFastJetsExtended");
+  else
+    iEvent.put(std::move(L1TrackFastJets), "L1TruthTrackFastJets");
+}
+
+void L1TruthTrackFastJetProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
```

Comment:
Please add a default description for the parameters in your module.

Reply:
Done.